### ISSUE TITLE
Support for comparing two properties

### ISSUE THREAD

**Issue by paw3lx** [GooRiOn/Valit#114]
I think it could be nice to have ability to compare two properties of an object:

`.Ensure(m => m.Password, _ => _.IsEqualTo(m => m.RePassword))`

What do you think about it?

**Comment by GooRiOn** [comment]
Yup, that looks cool. Do you want to create an overload for each binary operation, or just ``IsEqualTo()`` ?

**Comment by paw3lx** [comment]
It will be great to create overload for `IsLessThan()` and  `IsGreaterThan()` too.

**Comment by arekbal** [comment]
@paw3lx That would be also useful with BeginDate/EndDate/BeginTime/EndTime validations where one have to predate the other.
But, wouldn't adding another overload to Ensure() with another argument(model) achieve the same with somehow cleaner syntax? Like
`.Ensure(m => m.RePassword, (_, m) => _.IsEqualTo(m.Password))`

Or both could be made available but I think promoting single 'best' way of doing stuff is the right idea.

BTW. I don't like underscore arg names in our code and in examples. Because: "That is a convention used when you don't care about the parameter.".
https://stackoverflow.com/questions/2778362/what-does-this-mean-in-lambda-expressions

**Comment by paw3lx** [comment]
@GooRiOn What do you think about @arekbal syntax?

**Comment by GooRiOn** [comment]
I'd go for Arek's suggestion mostly because that is fairly simple to implement and does what it should. I also agree that we should promote "best way" of doing something instead of "couple fine" ways. The only problem I have is that the old syntax will be depricated (at the second version) :D 

On the one hand we don't have a lot of "fame" right now so "the soon the better". But on the other, it could piss off all folks that use that.

@dbarwikowski I summon you !

**Comment by dbarwikowski** [comment]
I prefer @paw3lx syntax. I think it's more human friendly and easier to remember.

@arekbal what would you suggest to use instead of '_'?

**Comment by arekbal** [comment]
'x', 'a', 'v' ?

**Comment by dbarwikowski** [comment]
'()' ?

**Comment by arekbal** [comment]
Let me just note that closures with reference interception would be a bit slower because these cannot be cached.
